lib/lrama/command.rb

```diff
@@ -5,65 +5,104 @@ class Command
+    rescue => e
+      message = e.message
+      message = message.gsub(/.+/, "\e[1m\\&\e[m") if Exception.to_tty?
+      abort message
+    end
+
+    def run
       Lrama::Report::Profile::CallStack.report(@options.profile_opts[:call_stack]) do
         Lrama::Report::Profile::Memory.report(@options.profile_opts[:memory]) do
-          _run
+          execute_command_workflow
         end
       end
     end
 
     private
 
-    def parser_options(argv)
-      @options = OptionParser.new.parse(argv)
+    def execute_command_workflow
+      @tracer.enable_duration
+      text = read_input
+      grammar = build_grammar(text)
+      states, context = compute(grammar)
+      render_reports(states) if @options.report_file
+      @tracer.trace(grammar)
+      render_diagram(grammar)
+      render_output(context, grammar)
+      validate_grammar(grammar, states)
+      @warnings.warn(grammar, states)
+    end
+
+    def read_input
+      text = @options.y.read
+      @options.y.close unless @options.y == STDIN
+      text
+    end
+
+    def build_grammar(text)
+      grammar =
+        Lrama::Parser.new(text, @options.grammar_file, @options.debug, @options.define).parse
+      merge_stdlib(grammar)
+      prepare_grammar(grammar)
+      grammar
     rescue => e
-      message = e.message
-      message = message.gsub(/.+/, "\e[1m\\&\e[m") if Exception.to_tty?
-      abort message
+      raise e if @options.debug
+      abort format_error_message(e.message)
     end
 
-    def _run
-      Report::Duration.enable if @options.trace_opts[:time]
+    def format_error_message(message)
+      return message unless $stderr.tty?
 
-      text = @options.y.read
-      @options.y.close if @options.y != STDIN
-      begin
-        grammar = Lrama::Parser.new(text, @options.grammar_file, @options.debug, @options.define).parse
-        unless grammar.no_stdlib
-          stdlib_grammar = Lrama::Parser.new(File.read(STDLIB_FILE_PATH), STDLIB_FILE_PATH, @options.debug).parse
-          grammar.insert_before_parameterizing_rules(stdlib_grammar.parameterizing_rules)
-        end
-        grammar.prepare
-        grammar.validate!
-      rescue => e
-        raise e if @options.debug
-        message = e.message
-        message = message.gsub(/.+/, "\e[1m\\&\e[m") if Exception.to_tty?
-        abort message
-      end
-      states = Lrama::States.new(grammar, trace_state: (@options.trace_opts[:automaton] || @options.trace_opts[:closure]))
+      message.gsub(/.+/, "\e[1m\\&\e[m")
+    end
+
+    def merge_stdlib(grammar)
+      return if grammar.no_stdlib
+
+      stdlib_text = File.read(STDLIB_FILE_PATH)
+      stdlib_grammar = Lrama::Parser.new(
+        stdlib_text,
+        STDLIB_FILE_PATH,
+        @options.debug
+      ).parse
+
+      grammar.insert_before_parameterizing_rules(stdlib_grammar.parameterizing_rules)
+    end
+
+    def prepare_grammar(grammar)
+      grammar.prepare
+      grammar.validate!
+    end
+
+    def compute(grammar)
```

Comment:
I prefer `#compute_states`.

Reply:
Fix in 76e59b9d30cadbcba51ebffc0973010b54b5bdd5